x-pack/plugin/rank-rrf/src/test/java/org/elasticsearch/xpack/rank/rrf/RRFRankDocTests.java

```diff
@@ -46,34 +51,46 @@ protected RRFRankDoc createTestInstance() {
 
     @Override
     protected RRFRankDoc mutateInstance(RRFRankDoc instance) throws IOException {
-        RRFRankDoc mutated = new RRFRankDoc(instance.doc, instance.shardIndex, instance.positions.length);
-        mutated.score = instance.score;
-        mutated.rank = instance.rank;
-        System.arraycopy(instance.positions, 0, mutated.positions, 0, instance.positions.length);
-        System.arraycopy(instance.scores, 0, mutated.scores, 0, instance.positions.length);
-        mutated.rank = mutated.rank == NO_RANK ? randomIntBetween(1, 10000) : NO_RANK;
+        int doc = instance.doc;
         if (rarely()) {
-            int ri = randomInt(mutated.positions.length - 1);
-            mutated.positions[ri] = mutated.positions[ri] == NO_RANK ? randomIntBetween(1, 10000) : NO_RANK;
+            doc = randomNonNegativeInt();
         }
-        if (rarely()) {
-            int ri = randomInt(mutated.positions.length - 1);
-            mutated.scores[ri] = randomFloat();
+        int shardIndex = instance.shardIndex;
+        if (frequently()) {
+            shardIndex = shardIndex == -1 ? randomNonNegativeInt() : -1;
         }
-        if (rarely()) {
-            mutated.doc = randomNonNegativeInt();
+        float score = instance.score;
+        if (frequently()) {
+            score = randomFloat();
         }
-        if (rarely()) {
-            mutated.score = randomFloat();
+        int rankConstant = instance.rankConstant;
+        if (frequently()) {
+            rankConstant = randomIntBetween(1, 100);
         }
+        int rank = instance.rank;
         if (frequently()) {
-            mutated.shardIndex = mutated.shardIndex == -1 ? randomNonNegativeInt() : -1;
+            rank = rank == NO_RANK ? randomIntBetween(1, 10000) : NO_RANK;
         }
+        int queries = instance.positions.length;
+        int[] positions = new int[queries];
+        float[] scores = new float[queries];
+        for (int i = 0; i < queries; i++) {
+            if (rarely()) {
+                positions[i] = instance.positions[i] == NO_RANK ? randomIntBetween(1, 10000) : NO_RANK;
+            } else {
+                positions[i] = instance.positions[i];
+            }
+            if (rarely()) {
+                scores[i] = randomFloat();
+            } else {
+                scores[i] = instance.scores[i];
+            }
+        }
+        RRFRankDoc mutated = new RRFRankDoc(doc, shardIndex, queries, rankConstant);
```

Comment:
Given that all the mutations are flagged with randomness, this will rarely return something that isn't mutated.

I would instead do a `switch(4) {case 0: <mutate doc>, case 1: <mutate shard index> ....` to ensure its changed on every call. Otherwise we will have flaky test failures.